packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -2675,47 +2681,47 @@ class WidgetInspector extends StatefulWidget {
 class _WidgetInspectorState extends State<WidgetInspector>
     with WidgetsBindingObserver {
 
-  _WidgetInspectorState() : selection = WidgetInspectorService.instance.selection;
+  _WidgetInspectorState();
 
   Offset? _lastPointerLocation;
 
-  final InspectorSelection selection;
+  InspectorSelection get selection =>
+      WidgetInspectorService.instance.selection;
 
-  /// Whether the inspector is in select mode.
-  ///
-  /// In select mode, pointer interactions trigger widget selection instead of
-  /// normal interactions. Otherwise the previously selected widget is
-  /// highlighted but the application can be interacted with normally.
-  bool isSelectMode = true;
+  bool get isSelectMode =>
+      WidgetInspectorService.instance.isSelectMode.value;
+  set isSelectMode(bool value) =>
+      WidgetInspectorService.instance.isSelectMode.value = value;
 
   final GlobalKey _ignorePointerKey = GlobalKey();
 
   /// Distance from the edge of the bounding box for an element to consider
   /// as selecting the edge of the bounding box.
   static const double _edgeHitMargin = 2.0;
 
-  InspectorSelectionChangedCallback? _selectionChangedCallback;
   @override
   void initState() {
     super.initState();
 
-    _selectionChangedCallback = () {
-      setState(() {
-        // The [selection] property which the build method depends on has
-        // changed.
-      });
-    };
-    WidgetInspectorService.instance.selectionChangedCallback = _selectionChangedCallback;
+    WidgetInspectorService.instance.selection.addListener(_selectionInformationChanged);
+    WidgetInspectorService.instance.isSelectMode
+        .addListener(_selectionInformationChanged);
   }
 
   @override
   void dispose() {
-    if (WidgetInspectorService.instance.selectionChangedCallback == _selectionChangedCallback) {
-      WidgetInspectorService.instance.selectionChangedCallback = null;
-    }
+    WidgetInspectorService.instance.selection
+        .removeListener(_selectionInformationChanged);
+    WidgetInspectorService.instance.isSelectMode
+        .removeListener(_selectionInformationChanged);
     super.dispose();
   }
 
+  void _selectionInformationChanged() => setState((){
+    // Triggers a rebuild of the WidgetInspector. For use when dependant
```

Comment:
nit: more idiomatic: load the current value of the selection into local fields here.
e.g.
selection = instance.selection.value;
selectMode = instance.isSelectMode.value
that way you don't need a comment. Downside is you will need to sync the initial value in initState.